src/rendering/framebufferrenderer.cpp

```diff
@@ -66,955 +66,963 @@ namespace {
+
+            ghoul::Dictionary dict;
+            dict.setValue("rendererData", _rendererData);
+            //dict.setValue("fragmentPath", fsPath);
+            dict.setValue("id", data.id);
+            std::string helperPath = caster->helperPath();
+            ghoul::Dictionary helpersDict;
+            if (helperPath != "") {
+                helpersDict.setValue("0", helperPath);
+            }
+            dict.setValue("helperPaths", helpersDict);
+            //dict.setValue("deferredcastPath", caster->getDeferredcastPath());
 
-        ghoul::Dictionary dict;
-        dict.setValue("rendererData", _rendererData);
-        dict.setValue("fragmentPath", fsPath);
-        dict.setValue("id", data.id);
-        std::string helperPath = raycaster->getHelperPath();
-        ghoul::Dictionary helpersDict;
-        if (!helperPath.empty()) {
-            helpersDict.setValue("0", helperPath);
-        }
-        dict.setValue("helperPaths", helpersDict);
-        dict.setValue("raycastPath", raycaster->getRaycastPath());
+            _deferredcastData[caster] = data;
 
-        _raycastData[raycaster] = data;
+            try {
+                ghoul::Dictionary deferredDict = dict;
+                //deferredDict.setValue("getEntryPath", GetEntryOutsidePath);
+                _deferredcastPrograms[caster] = ghoul::opengl::ProgramObject::Build(
+                    "Deferred " + std::to_string(data.id) + " raycast",
+                    absPath(vsPath),
+                    absPath(deferredShaderPath),
+                    deferredDict);
+                using IgnoreError = ghoul::opengl::ProgramObject::IgnoreError;
+                _deferredcastPrograms[caster]->setIgnoreSubroutineUniformLocationError(
+                    IgnoreError::Yes
+                );
+                _deferredcastPrograms[caster]->setIgnoreUniformLocationError(
+                    IgnoreError::Yes
+                );
 
-        try {
-            _exitPrograms[raycaster] = ghoul::opengl::ProgramObject::Build(
-                "Volume " + std::to_string(data.id) + " exit",
-                vsPath,
-                absPath(ExitFragmentShaderPath),
-                dict
-            );
-        } catch (ghoul::RuntimeError e) {
-            LERROR(e.message);
-        }
-        try {
-            ghoul::Dictionary outsideDict = dict;
-            outsideDict.setValue("getEntryPath", absPath(GetEntryOutsidePath));
-            _raycastPrograms[raycaster] = ghoul::opengl::ProgramObject::Build(
-                "Volume " + std::to_string(data.id) + " raycast",
-                absPath(vsPath),
-                absPath(RaycastFragmentShaderPath),
-                outsideDict
-            );
-        } catch (ghoul::RuntimeError e) {
-            LERROR(e.message);
+                caster->initializeCachedVariables(*_deferredcastPrograms[caster]);
+            }
+            catch (ghoul::RuntimeError& e) {
```

Comment:
catch by `const ...&`

Does it make sense to continue if the building fails?

Reply:
I think it depends of the context. But I would prefer to let it as is because right now it helped a lot with new developers. Maybe when the code is more mature we can remove it?

Reply:
Mhh.. yeah, I guess in this case we don't already fail catastrophically, but the exception will be caught almost immediately outside anyway if it fails.